std/functional.d

```diff
@@ -260,6 +210,82 @@ unittest
     static assert(!is(typeof(binaryFun!FuncObj)));
 }
 
+/**
+Transforms a string representing an expression into a n-ary function.
+The number of arguments is given as $(argc).
+The string must either use symbol names $(D a-z) as the parameters or
+provide the symbols via the $(D parmNames) arguments.
+If $(D fun) is not a string, $(D naryFun) aliases itself away to $(D fun).
+*/
+
+template naryFun(alias fun, size_t argc)
+{
+    enum defaults = AliasSeq!(
+        "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n",
+        "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z");
+    static assert(argc < defaults.length, "cannot generate > 26 parameters");
+    alias naryFun = naryFun!(fun, defaults[0..argc]);
+}
+
+/// ditto
+template naryFun(alias fun, parmNames...)
+    if (allSatisfy!(isSomeString, typeof(parmNames)))
+{
+    static if (is(typeof(fun) : string))
+    {
+        // create a string containing "alias <name> = args[0];", for each arg
+        auto aliases()
+        {
+            import std.conv : to;
+
+            string result;
+
+            foreach(i, name ; parmNames)
+                result ~= "alias " ~ name ~ " = __args[" ~ i.to!string ~ "];";
+
+            return result;
+        }
+
+        static if (!fun._ctfeMatchNary(parmNames))
+        {
+            import std.traits, std.typecons, std.typetuple;
+            import std.algorithm, std.conv, std.exception, std.math, std.range, std.string;
+        }
+
+        auto naryFun(ElementTypes...)(auto ref ElementTypes __args)
+        {
+            mixin(aliases());
+            return mixin(fun);
+        }
+    }
+    else static if (needOpCallAlias!fun)
+    {
+        // Issue 9906
+        alias naryFun = fun.opCall;
+    }
+    else
+    {
+        alias naryFun = fun;
+    }
+}
+
+///
+unittest
+{
+    // generated param names
+    alias sum = naryFun!("a + b + c", 3);
+    assert(sum(1, 2, 3) == 6);
+
+    // explicit param names
+    alias prod = naryFun!("x * y * z * w", "x", "y", "z", "w");
+    assert(prod(2,3,4,5) == 120);
+
+    // library function invocation
+    import std.algorithm : min;
```

Comment:
This import is not used. The symbols available to string lambdas are defined by what std.functional imports, not what the caller module imports.